redisson/src/main/java/org/redisson/client/RedisClient.java

```diff
@@ -367,6 +367,11 @@ public void run() {
         return result;
     }
 
+    @Override
+    public int  hashCode() {
+        return uri.hashCode();
```

Comment:
this violates equals / hashCode contract

Reply:
I also dislike this idea because the load will not spread when you have multiple slaves. Instead it should be configurable somehow... like changing `ConcurrentHashMap` in `MasterSlaveConnectionManager` to `ConcurrentSkipListMap` and allowing the application to pass in a custom `Comparator`.